src/backend/dx12/src/native.rs

```diff
@@ -497,8 +574,68 @@ impl HalDescriptorPool<Backend> for DescriptorPool {
         })
     }
 
+    fn free_sets(&mut self, descriptor_sets: &[DescriptorSet]) {
+        for descriptor_set in descriptor_sets {
+            for binding_info in &descriptor_set.binding_infos {
+                if let Some(ref view_range) = binding_info.view_range {
+                    let HeapProperties { has_view, .. } = view_range.ty.heap_properties();
+                    if has_view {
```

Comment:
nit: let's just do `if view_range.ty.heap_properties().has_view {..}`
the destructuring code is not needed here, just adds noise